File: Medium/2390. Removing Stars From a String/Solution.java
class Solution {
    public String removeStars(String s) {
        Stack<Character> sChars = new Stack<>();
        
        for(int i =0;i<s.length();i++){
            if(s.charAt(i)=='*'){
                int index = sChars.size()-1;
                while (sChars !=null){
                    if(sChars.get(index)!='*'){
                        sChars.pop();
                        break;
                    }
                    sChars.pop();
                    index--;
                }
            }
            else{
                sChars.add(s.charAt(i));
            }
        }
      StringBuilder sb = new StringBuilder();
        for (Character ch : sChars) {
            sb.append(ch);
        }
        return  sb.toString();
    }
}

File: Medium/2390. Removing Stars From a String/Solution.cs
using System;
using System.Collections.Generic;
using System.Text;

public class Solution {
    public string RemoveStars(string s) {
        var stack = new Stack<char>();
        
        foreach (char c in s) {
            if (c == '*') {
                if (stack.Count > 0) {
                    stack.Pop();
                }
            } else {
                stack.Push(c);
            }
        }
        
        var sb = new StringBuilder();
        while (stack.Count > 0) {
            sb.Insert(0, stack.Pop());
        }
        
        return sb.ToString();
    }
}